wasm/src/lib.rs

```diff
@@ -28,23 +42,35 @@ pub const MAX_BALANCES: usize = 10;
 pub const MAX_ORDERS: usize = 4;
 
 // Substitute for SizedWalletShare::NUM_SCALARS
-pub const NUM_SCALARS: usize = 69;
+pub const NUM_SCALARS: usize = 70;
 
 /// The number of bytes in a cluster symmetric key
 pub const CLUSTER_SYMMETRIC_KEY_LENGTH: usize = 32;
 
 #[wasm_bindgen]
-// sk_root: Hex
-pub fn get_pk_root(sk_root: &str) -> JsValue {
-    let signing_key = hex_to_signing_key(sk_root).unwrap();
-    let pk_root = signing_key.verifying_key();
-    JsValue::from_str(&hex::encode(pk_root.to_encoded_point(false).as_bytes()))
+pub fn derive_sk_root_from_seed(seed: &str, nonce: u64) -> JsValue {
+    let root_key = derive_root_signing_key(seed).unwrap();
+
+    let sk_root = derive_sk_root(&root_key, Some(&Scalar::from(nonce))).unwrap();
+    let sk_root_biguint = BigUint::from_bytes_be(&sk_root.to_bytes_be());
+    // TODO: Should be able to .into() here
+    let sk_root_scalar: NonNativeScalar<2> = NonNativeScalar::from(&sk_root_biguint);
+    let sk_root_hex = nonnative_scalar_to_hex_string(&sk_root_scalar);
+    JsValue::from_str(&sk_root_hex)
 }
 
 #[wasm_bindgen]
-pub fn pk_root_scalars(sk_root: &str) -> Vec<JsValue> {
-    let signing_key = hex_to_signing_key(sk_root).unwrap();
-    let keychain = wrap_eyre!(derive_wallet_keychain(&signing_key)).unwrap();
+pub fn get_pk_root(seed: &str, nonce: u64) -> JsValue {
+    let sk_root = derive_sk_root_signing_key(&seed, Some(&Scalar::from(nonce))).unwrap();
+    let keychain = wrap_eyre!(derive_wallet_keychain(&sk_root)).unwrap();
```

Comment:
we can just derive the public key here so we don't derive the whole keychain needlessly